identity/app/form/PrivacyMapping.scala

```diff
@@ -108,7 +110,34 @@ case class PrivacyFormData(
   }
 }
 
-object PrivacyFormData {
+object PrivacyFormData extends SafeLogging {
+
+  /**
+    * Checks if a consents is on the library and logs if not
+    *
+    * @param consent Consent
+    * @return Boolean (false if consent does not exist)
+    */
+  def checkIfConsentExistsInModel(consent:Consent): Boolean = {
+    Try(Consent.wording(consent.id, consent.version)) match {
+      case Success(wording:ConsentWording) => true
+      case Failure(f) => {
+        LogMissingConsent(consent)
```

Comment:
not a fan of this side effect, not sure how to refactor it in a way that doesn't just bury it deeper though :(